### PR TITLE
add user detail page

### PR DESCRIPTION
**Pull request by Rick1Marques**

### CODE REVIEW

File: components/UserListings/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  return (
+    <>
+      <Flex gap="1.5rem" flexWrap="wrap" direction="column" width="90%">
+        {listings.map((listing) => {
+          return (
+            <ListingCard
+              {...listing}
+              key={listing._id}
+              name={starshipURL(listing.starship).name}
+              preis={listing.preis}
```
Comment (shrocket) [inline components/UserListings/index.js:19]: Please remember that writing your code in English will help other devs be able to read it! The "preis" is not a horrible example of German invading the code, as most people could probably figure out what it means, but, you know, be careful! :)

File: pages/api/users/[id].js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  await connect();
+  if (request.method === "GET") {
+    const user = await User.findById(request.query.id)
+      .populate("listings")
+      .populate("starships")
+      .exec();
+    if (!user) {
+      response.status(404).json({
+        message:
+          "Sorry, something went wrong, 404 page not found! Please try again later.",
```
Comment (shrocket) [inline pages/api/users/[id].js:14]: The phrase "404 page not found" reads like the error page isn't being found. Maybe just leave this phrase out as the "404" is more for the devs than the users.

File: pages/starships/[id].js [hunk 3]
```diff
@@ -82,7 +81,7 @@ export default function Starship() {
           <StarshipInfo label="Starship class:" value={starship_class} />
         </Flex>
       </Flex>
-      <StarshipSellerList img={img} name={name} id={_id} />
+      <StarshipSellersList img={img} name={name} starshipId={_id} />
     </Flex>
```
Comment (shrocket) [inline pages/starships/[id].js:85]: We (we looked at this as a team) don't fully understand the decision/s you've made to put some styling into components and some into the pages like this. We certainly don't (yet?) think of ourselves as professionals, but we think your code would be cleaner if you exported your styling to components and kept your code on each page to an easily-looked-at minimum.

**Review by shrocket — APPROVED** [review]
It would appear as though everything works the way it's supposed to (in terms of the code), but we found it a little difficult to look at and understand because there is so much styling on each page that could be put into components. Our recommendation would be to refactor, but this is just a recommendation and not a reason to not approve. (Great presentation, earlier! App looks great!)

File: components/UserListings/index.js (hunk 1 above)
Comment (Rick1Marques) [reply components/UserListings/index.js:19]: That was an error on the backend (ops! xD). We are going to change some things in the backend this week and we will solve this problem as well!

File: pages/starships/[id].js (hunk 3 above)
Comment (Rick1Marques) [reply pages/starships/[id].js:85]: Some of the components are very simple and so, we decided to keep some of them on the page. But you are right, some of them should be a extra component. We will invest more time at the end of the project to make the css optimally.

**Pull request merged**